src/bindings/python/flux/job.py

```diff
@@ -0,0 +1,63 @@
+import os
+import six
+import sys
+import errno
+
+from flux.wrapper import Wrapper, FunctionWrapper
+from flux.core.inner import raw as core_raw
+from _flux._job import ffi, lib
+
+class FutureFunctionWrapper(FunctionWrapper):
+    def __init__(self, *args, **kwargs):
+        super(FutureFunctionWrapper, self).__init__(*args, **kwargs)
+
+    def __call__(self, calling_object, *args_in):
+        try:
+            return super(FutureFunctionWrapper,
+                         self).__call__(calling_object, *args_in)
+        except EnvironmentError as error:
+            exception_tuple = sys.exc_info()
+            try:
+                errmsg = ffi.string(core_raw.flux_future_error_string(future))
+                raise EnvironmentError(error.errno, errmsg)
+            except:
+                six.reraise(*exception_tuple)
+
+
+class JobWrapper(Wrapper):
+
+    def __init__(self):
+        super(JobWrapper, self).__init__(ffi, lib, prefixes=['flux_job_',])
+
+    def check_wrap(self, fun, name):
+        fun_type = self.ffi.typeof(fun)
+        add_handle = self.check_handle(name, fun_type)
+        return FutureFunctionWrapper(fun, name, fun_type, self.ffi,
+                                     add_handle=add_handle)
+
+RAW = JobWrapper()
+
+def submit_async(flux_handle, jobspec,
```

Comment:
I'm good with this as a first cut, we'll want to add some validation here, but I think we want to straighten that out elsewhere first.